File: src/services/codex-sdk-service.ts
import {
  createCodexSdkSession,
  listCodexSdkSessions,
  getCodexSdkSessionDetails,
  deleteCodexSdkSession,
  sendCodexSdkUserMessage,
} from '../core/codex-sdk-sessions.js';
import type { CodexSdkEvent, CodexSdkSessionSummary } from '../types/codex-sdk.js';

export interface CodexSdkSessionDetail {
  session: CodexSdkSessionSummary;
  events: CodexSdkEvent[];
}

export class CodexSdkService {
  constructor(private readonly workdir: string) {}

  async listSessions(input: { org: string; repo: string; branch: string }): Promise<CodexSdkSessionSummary[]> {
    return listCodexSdkSessions({ ...input, workdir: this.workdir });
  }

  async createSession(input: {
    org: string;
    repo: string;
    branch: string;
    label?: string;
  }): Promise<CodexSdkSessionDetail> {
    const result = await createCodexSdkSession({ ...input, workdir: this.workdir });
    return { session: result.summary, events: result.events };
  }

  async getSession(sessionId: string): Promise<CodexSdkSessionDetail | null> {
    const details = getCodexSdkSessionDetails(sessionId);
    return details ? { session: details.summary, events: details.events } : null;
  }

  async deleteSession(sessionId: string): Promise<void> {
    await deleteCodexSdkSession(sessionId);
  }

  async sendMessage(sessionId: string, text: string): Promise<void> {
    await sendCodexSdkUserMessage(sessionId, text);
  }
}

export function createCodexSdkService(workdir: string): CodexSdkService {
  return new CodexSdkService(workdir);
}

File: mobile/ios/AgentrixMobile/Services/CodexSdkService.swift
import Foundation

final class CodexSdkService {
    private let api: AgentrixAPIClient

    init(api: AgentrixAPIClient) {
        self.api = api
    }

    func listSessions(org: String, repo: String, branch: String) async throws -> [CodexSdkSessionSummary] {
        struct Response: Decodable {
            let sessions: [CodexSdkSessionSummary]
        }
        var components = URLComponents()
        components.queryItems = [
            URLQueryItem(name: "org", value: org),
            URLQueryItem(name: "repo", value: repo),
            URLQueryItem(name: "branch", value: branch)
        ]
        let query = components.percentEncodedQuery.map { "?\($0)" } ?? ""
        let response: Response = try await api.request("/api/codex-sdk/sessions\(query)")
        return response.sessions
    }

    func createSession(org: String, repo: String, branch: String, label: String?) async throws -> CodexSdkSessionDetail {
        struct Payload: Encodable {
            let org: String
            let repo: String
            let branch: String
            let label: String?
        }
        return try await api.request(
            "/api/codex-sdk/sessions",
            method: .post,
            body: Payload(org: org, repo: repo, branch: branch, label: label)
        )
    }

    func deleteSession(sessionId: String) async throws {
        try await api.requestVoid(
            Self.sessionPath(sessionId),
            method: .delete
        )
    }

    func makeWebSocketTask(sessionId: String) -> URLSessionWebSocketTask {
        let url = api.config.codexSdkWebSocketURL(sessionId: sessionId)
        return api.session.webSocketTask(with: url)
    }

    private static func sessionPath(_ sessionId: String) -> String {
        let encoded = sessionId.addingPercentEncoding(withAllowedCharacters: .urlPathAllowed) ?? sessionId
        return "/api/codex-sdk/sessions/\(encoded)"
    }
}
